content/blog/2025-06-25-expanding-catalog.mdx

```diff
@@ -0,0 +1,97 @@
+---
+title: "GSoC'25: Expanding the Unikraft Software Support Ecosystem"
+description: |
+  This GSoC project aims to expand Unikraft’s software support by adding new applications via binary-compatibility mode into the application catalog.
+publishedDate: 2025-06-25
+image: /images/gsoc25.jpeg
+authors:
+- Prasoon Kumar
+tags:
+- gsoc
+- gsoc25
+- catalog
+- sqlite
+- mosquitto
+- prometheus
+- consul
+---
+
+## Project Overview
+
+Unikraft makes it easy to run existing applications in two ways:
+1. **Musl libc support**
+
+1. [**Binary-compatibility mode**](https://unikraft.org/docs/concepts/compatibility):
+  Unikraft can run unmodified Linux ELF binaries by using an ELF loader app and a system-call shim layer.
+  When a binary-compatible unikernel starts, the app-elfloader parses and maps the ELF segments into memory, then transfers control to the application entry point.
+  Whenever the application invokes a Linux system call, the syscall shim intercepts the call number and routes it to the corresponding Unikraft handler.
+  Unimplemented calls return `ENOSYS`, which many applications gracefully stub or fake, thanks to the close adherence to Linux’s ABI on x86_64 (KVM/QEMU/Firecracker)
+
+## Adding an application in binary-compatibility mode
+
+Though the full procedure is detailed in the Unikraft Docs (see [Adding Applications to the Catalog](https://unikraft.org/docs/contributing/adding-to-the-app-catalog)), here are a few additional findings:
+
+1. **Find supported syscalls**
+    - Search for syscall macros in the Unikraft source (see [Syscall Shim Layer](https://unikraft.org/docs/internals/syscall-shim)).
+    - Or look for `UK_PROVIDED_SYSCALLS-` entries in `Makefile.uk` files of Unikraft libraries.
+1. **Discover the application’s syscalls**
+    - Run the application under strace inside a container.
+    - Save the trace output and extract the unique syscall list.
```

Comment:
How to extract the unique syscall list? Did you implement a custom script for it? If so, I think it is worth mentioning here